### PR TITLE
Add proper exception handling

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?
Add proper exception handling to make it more user friendly
## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
Unit tests and local tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1568

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/V1LayerConverter.scala [hunk 1]
```diff
@@ -42,7 +42,8 @@ class V1LayerConverter[T: ClassTag](implicit ev: TensorNumeric[T]) extends Conve
     val weightBlob = getBlob(layer, 0).get
     val biasBlob = getBlob(layer, 1)
     if (!biasBlob.isDefined) {
-      throw new RuntimeException(s"${getLayerName(layer)} without bias is not supported now")
+      throw new CaffeConversionException(s"${getLayerName(layer)} " +
+        s"without bias is not supported now")
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/V1LayerConverter.scala]: It should be supported now

**Comment by SeaOfOcean** [comment]
What if there is layer in prototxt definition but not in binary file?

**Comment by wzhongyuan** [comment]
@SeaOfOcean  Refined on top of your comments. for the question 'what if there is layer in prototxt definition but not in binary',  I just  print a warning to user, the reason was maybe someone modify the prototxt and add some more layers which do not have weights or weight is not necessary on loading (maybe continue training after loading)

**Comment by SeaOfOcean** [comment]
yes it is reasonable for user to modify some layers.
The problem is that if they append a new linear or convolution (layers with parameters),
since you need to use its weight blob to determine the nOutput, it will throw exception.

**Comment by wzhongyuan** [comment]
@SeaOfOcean  Makes sense,  we should do sanity check where needed, will come back with that shortly

**Comment by wzhongyuan** [comment]
@SeaOfOcean  pls review the updates

**Comment by wzhongyuan** [comment]
PR validation passed http://172.168.2.101:8080/job/BigDL-PR-V/528/

**Pull request merged**